src/core/federation.ts

```diff
@@ -1347,6 +1403,44 @@ export class FederatedApp extends App {
     return true;
   }
 
+  /**
+   * @override
+   * Iterate over all reactions in the reaction queue and execute them.
+   * If a reaction's priority is less than MLAA, stop executing and return.
+   * @returns Whether every reactions at this tag are executed.
+   */
+  protected _react(): boolean {
```

Comment:
@lhstrh I removed the call of `_updateMaxLevel` here and the function is called only when it is necessary (please refer to the commit [a413a9f](https://github.com/lf-lang/reactor-ts/pull/198/commits/a413a9f9da76572e0a570b06358e11ee0274204a)).

One question is, if I remember correctly, you also wanted to divide this `_react` function, right? Nonetheless, I don't know how to separate this function. Because this function has a simple role of doing react until the reaction of `_reactionQ` is allowed to execute. 

A possible separation is, pushing executable reactions into a new list (in the new function) and then processing them (in the `_react()`). However, I and @hokeun think that the current state is also clear. What do you think about?

Reply:
I agree that this is fine. I think that maybe I didn't realize that this was overriding the `_react` function of `App`. That said, it looks like there are currently some federation-related functions in `App` that should probably be moved into `FederatedApp`. But this can be done in a separate PR...